addons/web/tests/test_db_manager.py

```diff
@@ -119,3 +137,26 @@ def test_database_duplicate(self):
         self.assertEqual(res.status_code, 303)
         self.assertIn('/web/database/manager', res.headers['Location'])
         self.assertDbs([])
+
+        # check if user is automatically logout after trying to access a deleted database.
+        # NOTE: the current behaviour is that the next request will crash, returning
+        # a response 500, but will also logout the user.
+        # If this changes and all sessions are cleaned with the database, we could remove
+        # the request on /web before checking the session content
+
+        # request /web to invalidate the session on a dropped database
+        with mute_logger('werkzeug'):
+            self.other_session.get(self.url('/web'))
+            #self._logger = logging.getLogger(__name__)  # pylint: disable=attribute-defined-outside-init
+            # the raised OperationalError will reach werkzeug and may be logged after the end of the request,
+            # outside the mute logger. We need to wait for remaining request to avoid that
+            HttpCase._wait_remaining_requests(self)
```

Comment:
Maybe `_wait_remaining_requests` could be converted to a free function, to which a logger would be passed, for less ugly? And that way `self._logger` wouldn't be *necessary* (though it could still be useful).

Reply:
Yes I thinked about that but I didn't want to change the behaviour or using a specific logger giving a clear information on the test causing the error.

And having a logger like that on the Base Case looked like a good idea. I agree with the `( it could still be useful).`

I still extracted _wait_remaining_requests outside HTTPCase with an optionnal logger parametter, not sure it is better like that. My use case to use it outside an HTTPCase is not usual.

Reply:
> I still extracted `_wait_remaining_requests` outside HTTPCase with an optionnal logger parametter, not sure it is better like that. My use case to use it outside an HTTPCase is not usual.

I think it's better, strictly speaking `_wait_remaining_requests` doesn't depend on any part of the test case and while it's definitely test-related (mostly) it makes sense as a separate utility. Maybe remove the `_` prefix as well?